### ISSUE TITLE
Support --no-verify in pre-commit hooks - feature request

### ISSUE THREAD

**Issue by dekelb** [nvbn/thefuck#1096]
`thefuck --version`:

    The Fuck 3.30 using Python 3.8.2 and ZSH 5.3

Your system (Debian 7, ArchLinux, Windows, etc.):

    MacOS

How to reproduce the bug:

    Have a repo with a pre-commit hook, make sure this hook fails, run `git commit`
    The output of the git cmd would be something like this:

    npm ERR! A complete log of this run can be found in:
    npm ERR!     /Users/user/.npm/_logs/2020-05-15T12_44_24_633Z-debug.log
    husky > pre-commit hook failed (add --no-verify to bypass)

In the above example I used https://github.com/typicode/husky#readme, but I assume any other pre-commit hook will work.

**Comment by divykj** [comment]
I'm on it.

**Comment by divykj** [comment]
Added a basic implementation for bypassing failed git hooks (using --no-verify flag). Also added this rule for 3 commands which can fail due to a hook, namely `am`, `commit` and, `push`.

Can @dekelb or someone else please test this, and see if it works as intended.

Thank you.

**Comment by dekelb** [comment]
Seems like it doesn't work (sorry it took some time to test it).
```
➜  REPO git:(feature/rows-dnd) ✗ git commit demo/demo.js -m "working example"
husky > pre-commit (node v13.11.0)
....
....
✖ 5 problems (5 errors, 0 warnings)

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! package@1.5 eslint: `eslint src/** -c ./configs/.eslintrc`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the package@1.5 eslint script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/{me}/.npm/_logs/2020-06-02T22_51_40_019Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! package@1.5 lint: `npm run eslint && npm run tsc`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the package@1.5 lint script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/{me}/.npm/_logs/2020-06-02T22_51_40_036Z-debug.log
husky > pre-commit hook failed (add --no-verify to bypass)
➜  REPO git:(feature/rows-dnd) ✗ fuck -y
​​​​​​​​​​git commit --amend
husky > pre-commit (node v13.11.0)
....
....
✖ 5 problems (5 errors, 0 warnings)

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! package@1.5 eslint: `eslint src/** -c ./configs/.eslintrc`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the package@1.5 eslint script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/{me}/.npm/_logs/2020-06-02T22_51_48_141Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! package@1.5 lint: `npm run eslint && npm run tsc`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the package@1.5 lint script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/{me}/.npm/_logs/2020-06-02T22_51_48_158Z-debug.log
husky > pre-commit hook failed (add --no-verify to bypass)
```

Regarding what I'm actually expecting here - since the command was `git commit demo/demo.js -m "working example"` the expected behaviour would be `git commit demo/demo.js -m "working example" --no-verify`

**Comment by scorphus** [comment]
`fuck -y` wouldn't work in this case. The response from a pre-commit hook may take way too long to produce useful output. Hence, this rule has to be output-independent. But then, it should have a lower priority, which would throw it to the back of the queue of suggestions.

**Comment by divykj** [comment]
Agreed @scorphus, these hooks can take a long time to execute.

I'll try to complete the PR asap. I'm a little busy right now.

**Comment by divykj** [comment]
@scorphus I think this was resolved with #1097 and should be closed.

**Comment by scorphus** [comment]
Thanks for contributing, @divykj <3
